### ISSUE TITLE
Remove generic argument from `Application`

### ISSUE THREAD

**Issue by lann** [fermyon/spin#414]
`Application<T>` currently always has `T = CoreComponent`. If there isn't a compelling reason for `T` to change, we should remove it.

**Comment by Mossaka** [comment]
that's what I observed yesterday and agree that everywhere `Application` is used, it always uses `CoreComponent`. I thought the reason for it is that we could potentially have `RemoteComponent` that points to wasm module in GitHub or other places...

**Comment by Mossaka** [comment]
I can work on this issue

**Comment by Mossaka** [comment]
Not sure which one is better

```rust
pub struct Application<T = CoreComponent> {
    /// General application information.
    pub info: ApplicationInformation,
    /// Configuration for the application components.
    pub components: Vec<T>,
    /// Configuration for the components' triggers.
    pub component_triggers: ComponentMap<TriggerConfig>,
    /// Application-specific configuration resolver.
    pub config_resolver: Option<Arc<Resolver>>,
}
```

or 

```rust
pub struct Application {
    /// General application information.
    pub info: ApplicationInformation,
    /// Configuration for the application components.
    pub components: Vec<CoreComponent>,
    /// Configuration for the components' triggers.
    pub component_triggers: ComponentMap<TriggerConfig>,
    /// Application-specific configuration resolver.
    pub config_resolver: Option<Arc<Resolver>>,
}
```

**Comment by lann** [comment]
Just remove it entirely I think to avoid confusion. If it is needed in the future again it is no harder to add back as a default.

**Comment by Mossaka** [comment]
> Just remove it entirely I think to avoid confusion. If it is needed in the future again it is no harder to add back as a default.

sure!

**Comment by lann** [comment]
Fixed by #430
